File: packages/stronghold_core/lib/models/requests/update_seminar_request.dart
/// Matches backend UpdateSeminarRequest exactly
class UpdateSeminarRequest {
  final String? topic;
  final String? speakerName;
  final DateTime? eventDate;
  final int? maxCapacity;

  const UpdateSeminarRequest({
    this.topic,
    this.speakerName,
    this.eventDate,
    this.maxCapacity,
  });

  Map<String, dynamic> toJson() {
    final map = <String, dynamic>{};
    if (topic != null) map['topic'] = topic;
    if (speakerName != null) map['speakerName'] = speakerName;
    if (eventDate != null) map['eventDate'] = eventDate!.toIso8601String();
    if (maxCapacity != null) map['maxCapacity'] = maxCapacity;
    return map;
  }
}

File: Stronghold/Stronghold.Application/DTOs/Request/UpdateSeminarRequest.cs
using System.ComponentModel.DataAnnotations;

namespace Stronghold.Application.DTOs.Request
{
    public class UpdateSeminarRequest
    {
        [StringLength(100, MinimumLength = 2, ErrorMessage = "Tema mora imati između 2 i 100 karaktera.")]
        public string? Topic { get; set; }

        [StringLength(100, MinimumLength = 2, ErrorMessage = "Ime govornika mora imati između 2 i 100 karaktera.")]
        public string? SpeakerName { get; set; }

        public DateTime? EventDate { get; set; }

        [Range(1, 10000, ErrorMessage = "Kapacitet mora biti između 1 i 10000.")]
        public int? MaxCapacity { get; set; }
    }
}
